mxcube3/routes/Queue.py

```diff
@@ -242,155 +234,222 @@ def queue_load_state():
-                except Exception:
-                    logging.getLogger('HWR').exception('[QUEUE] Queue error executing child entry with id: %s' % elem['QueueId'])
-    else:
-        #  not a sample so execute directly
-        logging.getLogger('HWR').info('[QUEUE] Queue executing entry with id: %s' % node_id)
-        if mxcube.queue.queue_hwobj.is_paused():
-            logging.getLogger('HWR').info('[QUEUE] Cannot execute, queue is paused. Waiting for unpause')
-            msg = {'Signal': 'QueuePaused',
-                   'Message': 'Queue execution paused',
-                   'State': 1}
-            socketio.emit('Queue', msg, namespace='/hwr')
-            mxcube.queue.queue_hwobj.wait_for_pause_event()
-
-        entry._view = Mock()  # associated text deps
-        entry._set_background_color = Mock()  # widget color deps
-        #  parent = int(node.get_parent()._node_id)
-        # this is a TaskGroup, so it is not in the parsed queue
-        parent_node = node.get_parent() 
-        #  go a level up,
-        # this is a TaskGroup for a Char, a sampleQueueEntry if DataCol
-        parent_node = parent_node.get_parent()  
-        if isinstance(parent_node, qmo.TaskGroup):
-            parent_node = parent_node.get_parent()
-        parent = int(parent_node._node_id)
-
-        mxcube.queue.last_queue_node.update({'id': node_id,
-                                             'sample': queue[parent]['SampleId']
-                                             })
-        # mxcube.queue.queue_hwobj.execute_entry = types.MethodType(Utils.my_execute_entry, mxcube.queue.queue_hwobj)
-        mxcube.queue.queue_hwobj.execute_entry(entry)
-        entry.set_enabled(False)
+    mxcube.queue.queue_hwobj.execute_entry(entry)
 
     msg = {'Signal': 'QueueStopped',
            'Message': 'Queue execution stopped',
            'State': 1}
+
     socketio.emit('Queue', msg, namespace='/hwr')
 
     return Response(status=200)
 
 
 @mxcube.route("/mxcube/api/v0.1/queue", methods=['POST'])
-def add_sample():
-    '''
-    Add a sample to the queue.
-        :request Content-Type: application/json, {"SampleId": sampleId},
-        where sampleId is sample location (eg '1:01')
-        :response Content-Type: application/json, {"QueueId": node_id,
-            "SampleId": sampleId}, where sampleId is the same as the
-            caller id (eg '1:01') and node_id an integer which is used
-            for refering to this element from now onwards.
-        :statuscode: 200: no error
-        :statuscode: 409: sample could not be added, possibly because
-            it already exist in the queue
-        :example request:   * POST http://host:port/mxcube/api/v0.1/queue
-                            * Content-Type: application/json
-                            * {"SampleId": "1:07"}
-    '''
-    params = request.data
-    params = json.loads(params)
-    sample_id = params['SampleId']
+def queue_set():
```

Comment:
This is a O(n) operation (instead of O(1) as it was before) and it seems to be following the linear formula (y=kx) where y is milliseconds and x is number of tasks such as datacollection. So in other words with every task that is added there will be an additional X milliseconds of wait. 

This could be quite bad, I have done some testing with mxcube in build mode and it becomes quite unusable when the queue grows. I have talked to Jie and it is not at all impossible to have several hundred tasks in a queue after a long shift.